### PR TITLE
Add media components and data|code|media subtypes

### PR DESCRIPTION
**Pull request by shadrach-tayo**
## Description of the Problem / Feature
- Extend Component type models to support Media and more subtypes for existing models

### CODE REVIEW

File: desci-models/src/ResearchObject.ts [hunk 1]
```diff
@@ -138,6 +139,25 @@ export enum ResearchObjectComponentDocumentSubtype {
...
+  SOFTWARE_PACKAGE = "software-package",
+  CONTAINER = "container",
+}
+
+export enum MediaComponentSubtype {
+  COMMUNITY_DISCUSSION = "community-discussion",
+  VIDEO_RESOURCE = "video-resource",
+  EXTERNAL_REPOSITORY = "external-repository",
+  IMAGE_RESOURCE = "image-resource",
+  API_ENDPOINT = "api-endpoint",
```
Comment (hubsmoke) [inline desci-models/src/ResearchObject.ts:158]: external link

File: desci-models/src/ResearchObject.ts [hunk 2]
```diff
@@ -138,6 +139,25 @@ export enum ResearchObjectComponentDocumentSubtype {
...
+export enum CodeComponentSubtype {
+  JUPYTER_NOTEBOOK = "jupyter-notebook",
+  SOFTWARE_PACKAGE = "software-package",
+  CONTAINER = "container",
+}
+
+export enum MediaComponentSubtype {
+  COMMUNITY_DISCUSSION = "community-discussion",
+  VIDEO_RESOURCE = "video-resource",
+  EXTERNAL_REPOSITORY = "external-repository",
```
Comment (hubsmoke) [inline desci-models/src/ResearchObject.ts:156]: code

File: desci-models/src/ResearchObject.ts [hunk 3]
```diff
@@ -138,6 +139,25 @@ export enum ResearchObjectComponentDocumentSubtype {
...
+}
+
+export enum CodeComponentSubtype {
+  JUPYTER_NOTEBOOK = "jupyter-notebook",
+  SOFTWARE_PACKAGE = "software-package",
+  CONTAINER = "container",
+}
+
+export enum MediaComponentSubtype {
+  COMMUNITY_DISCUSSION = "community-discussion",
```
Comment (hubsmoke) [inline desci-models/src/ResearchObject.ts:154]: external link

**Comment by hubsmoke** [comment]
@shadrach-tayo can you explain why this change is necessary and if product still wants to move forward with this / what feature does this enable?

**Comment by shadrach-tayo** [comment]
> @shadrach-tayo can you explain why this change is necessary and if product still wants to move forward with this / what feature does this enable?

To have more types exposed as we support more content types

**Comment by m0ar** [comment]
@shadrach-tayo Is this stale/irrelevant as of the recent work of @Electromorphous ?

**Comment by hubsmoke** [comment]
@shadrach-tayo closing this unless it's still relevant?

**Pull request closed without merging**